autogen/agentchat/contrib/gpt_assistant_agent.py

```diff
@@ -133,7 +141,8 @@ def _invoke_assistant(
             self._openai_client.beta.threads.messages.create(
                 thread_id=assistant_thread.id,
                 content=message["content"],
-                role=message["role"],
+                # https://platform.openai.com/docs/api-reference/messages/createMessage - only user role accepted to create message in thread
+                role="user",
```

Comment:
Yes. If it is not user role, it may indicate that there is something wrong with our messaging mechanism. It is best to report an error.

Reply:
From the sender's perspective, the role of the message sent is assistant. From the receiver's perspective, the role of the message received is user. The autogen mechanism has already guaranteed this. So no modification is needed. Or did you encounter an error and need to correct it?

Reply:
makes sense, tested locally again and it seemed to work will keep an eye and open seperate issue if it happens again